src/stacks-hierarchy/Root_PurchaseConfirmationScreen/Root_PurchaseConfirmationScreen.tsx

```diff
@@ -297,61 +295,101 @@ export const Root_PurchaseConfirmationScreen: React.FC<Props> = ({
-                }}
-                loading={reserveMutation.isLoading}
-              />
-            ) : (
-              <Button
-                expanded={true}
-                interactiveColor={interactiveColor}
-                text={t(PurchaseConfirmationTexts.choosePaymentMethod.text)}
-                disabled={!!offerError}
-                accessibilityHint={t(
-                  PurchaseConfirmationTexts.choosePaymentMethod.a11yHint,
-                )}
-                onPress={() => {
-                  analytics.logEvent('Ticketing', 'Confirm purchase clicked', {
-                    mode: params.mode,
-                  });
-                  selectPaymentMethod();
-                }}
-                testID="choosePaymentMethodButton"
-                ref={onCloseFocusRef}
-              />
-            )}
-          </View>
-        )}
+        <PaymentButton
+          isSearchingOffer={isSearchingOffer}
+          isOfferError={!!offerError}
+          mode={params.mode}
+          onCloseFocusRef={onCloseFocusRef}
+          totalPrice={totalPrice}
+          reserveStatus={reserveMutation.status}
+          paymentMethod={paymentMethod}
+          onSelectPaymentMethod={selectPaymentMethod}
+          onGoToPayment={goToPayment}
+        />
       </View>
     </FullScreenView>
   );
 };
 
+type PaymentButtonProps = {
+  isSearchingOffer: boolean;
+  isOfferError: boolean;
+  totalPrice: number;
+  mode: 'TravelSearch' | 'Ticket' | undefined;
+  onCloseFocusRef: RefObject<any>;
+  reserveStatus: MutationStatus;
+  paymentMethod: PaymentMethod | undefined;
+  onSelectPaymentMethod: () => void;
+  onGoToPayment: () => void;
+};
+const PaymentButton = ({
+  isSearchingOffer,
+  isOfferError,
+  totalPrice,
+  mode,
+  onCloseFocusRef,
+  reserveStatus,
+  paymentMethod,
+  onSelectPaymentMethod,
+  onGoToPayment,
+}: PaymentButtonProps) => {
+  const {theme} = useThemeContext();
+  const {t, language} = useTranslation();
+  const analytics = useAnalyticsContext();
+  const totalPriceString = formatNumberToString(totalPrice, language);
+
+  if (isSearchingOffer)
+    return (
+      <ActivityIndicator
+        size="large"
+        color={theme.color.foreground.dynamic.primary}
+        style={{margin: theme.spacing.medium}}
+      />
+    );
+
+  if (!paymentMethod)
+    return (
+      <Button
+        expanded={true}
+        interactiveColor={theme.color.interactive[0]}
+        text={t(PurchaseConfirmationTexts.choosePaymentMethod.text)}
+        disabled={!!isOfferError}
+        accessibilityHint={t(
+          PurchaseConfirmationTexts.choosePaymentMethod.a11yHint,
+        )}
+        onPress={() => {
+          analytics.logEvent('Ticketing', 'Confirm purchase clicked', {
+            mode: mode,
+          });
+          onSelectPaymentMethod();
+        }}
+        testID="choosePaymentMethodButton"
+        ref={onCloseFocusRef}
+      />
+    );
+
+  return (
+    <Button
+      expanded={true}
```

Comment:
```suggestion
      expanded
```

Reply:
The app code mostly has `={true}` set for props like this. I'm open to changing that convention as bigger refactor (maybe an ESLint rule for one or the other), but for now I'll stick with this.

Reply:
Ah, TIL! Let's stick to that convention then.